mathesar_ui/src/systems/connections/AddConnection.svelte

```diff
@@ -0,0 +1,442 @@
+    modifiable: boolean;
+  }
+  function getCredentialsStrategyLabel(s: CredentialsStrategy) {
+    return s.reuse
+      ? $_('reuse_credentials_from_known_connection')
+      : $_('enter_new_credentials');
+  }
+
+  interface UserStrategy {
+    /** We want to create a new PostgreSQL user */
+    create: boolean;
+    /** The UI user can modify the strategy */
+    modifiable: boolean;
+  }
+  function getUserStrategyLabel(s: UserStrategy) {
+    return s.create ? $_('create_new_pg_user') : $_('use_existing_pg_user');
+  }
+
+  type InstallationSchema = SampleDataSchemaIdentifier | 'internal';
+  const installationSchemaOptions: InstallationSchema[] = [
+    'internal',
+    ...sampleDataOptions,
+  ];
+  const installationSchemaLabels: Record<InstallationSchema, string> = {
+    // These strings are not translated because these are the names that will
+    // appear in the Mathesar UI after installation.
+    internal: 'mathesar_types, __msar, msar',
+    library_management: 'Library Management',
+    movie_collection: 'Movie Collection',
+  };
+  const installationSchemaHelp: Record<InstallationSchema, string> = {
+    internal: $_('internal_schema_help'),
+    library_management: $_('sample_data_library_help'),
+    movie_collection: $_('sample_data_movies_help'),
+  };
+
+  export let onCancel: () => void;
+  export let onSuccess: (c: Connection) => void;
+
+  $: availableConnectionsToReuse = $generalConnections;
+  $: defaultConnectionToReuse =
+    pickDefaultGeneralConnection($generalConnections);
+  $: namedConnections = connectionsStore.connections;
+  $: connectionNames = new Set(map(([, c]) => c.nickname, $namedConnections));
+  $: someUserDbConnectionsExist = $generalConnections.some(
+    (c) => c.type === 'user_database',
+  );
+  $: defaultDatabaseName = someUserDbConnectionsExist ? '' : 'mathesar';
+  $: credentialsStrategy = requiredField<CredentialsStrategy>(
+    defaultConnectionToReuse
+      ? { reuse: true, modifiable: true }
+      : { reuse: false, modifiable: false },
+  );
+  $: databaseName = requiredField(defaultDatabaseName);
+  $: createDatabase = requiredField(true);
+  $: installationSchemas = requiredField<InstallationSchema[]>(['internal']);
+  $: connectionNickname = requiredField('', [uniqueWith(connectionNames)]);
+  $: connectionToReuse = requiredField(defaultConnectionToReuse);
+  $: host = requiredField('localhost');
+  $: port = requiredField(5432);
+  $: existingUserName = requiredField('');
+  $: existingPassword = requiredField('');
+  $: availableBootstrapConnections = $generalConnections.filter(
+    (c) => c.connection.host === $host && c.connection.port === $port,
+  );
+  $: defaultBootstrapConnection = pickDefaultGeneralConnection(
+    availableBootstrapConnections,
+  );
+  $: userStrategy = requiredField<UserStrategy>(
+    defaultBootstrapConnection
+      ? { create: false, modifiable: true }
+      : { create: false, modifiable: false },
+  );
```

Comment:
@seancolsen 

This logic is making me question the entire UX for Case C (Creating a new Postgres user), as I review it. 
1. If we're going to check for the same host name and port with the list of connections at hand to enable creating a new user, why does the user need to input the host name and port instead of selecting from the list?
1. If the user is going to input the host and the port, why would they want to select an existing user account to create a new one for Case C? Wouldn't this make more sense with Case A?

I understand this should have been discussed earlier, but the UX is confusing as I attempt to use it, and we should discuss this once again. cc: @ghislaineguerin, @kgodey

Reply:
Here's a screen recording describing the issue (Please enable audio).

https://github.com/mathesar-foundation/mathesar/assets/11850603/9e0af15d-ea52-40fd-92b7-3f58723dbc91

Reply:
We briefly [discussed](https://matrix.to/#/!kpLRqiyQwsndKhBIvd:matrix.mathesar.org/$elSq90BM9oH6HC3YeKwUgf_7GUkQ_WtD9bHylmSNZQw?via=matrix.mathesar.org&via=matrix.org) this on Matrix today. Next steps are for us to have more discussion after holiday break.

Reply:
We discussed this today at our maintainers meeting and decided to remove case C entirely. I'll be working on that next.

Reply:
I removed case C in e3d0ceeb504ffc63eae403f4deea21fc52345de3